spec/lang/representation.md

```diff
@@ -371,8 +401,10 @@ Similarly, on `Pointer` we say that adding provenance makes it more defined:
 ```rust
 impl<Provenance> DefinedRelation for Pointer<Provenance> {
     fn le_defined(self, other: Self) -> bool {
-        self.addr == other.addr &&
-            match (self.provenance, other.provenance) {
+        self.thin_pointer.addr == other.thin_pointer.addr &&
```

Comment:
Please don't skip over abstractions. Instead, add an `impl DefinedRelation for ThinPointer` and use that from here.